### PR TITLE
[Lock] Added paragraph about auto release

### PR DESCRIPTION
**Pull request by Nyholm**
This will fix https://github.com/symfony/symfony/issues/40325

@twisted1919 is this paragraph helpful? Should I add something more?

### CODE REVIEW

**Comment by twisted1919** [comment]
I think there should be a note for PCNTL as well, something like: 
> Please pay special attention when you create a lock and then you call `pcntl_fork` to create a new process. Resulted process will release the lock when it finishes, leaving the main process unlocked. To prevent this, make sure you do not autoi-release the lock and instead, you release it manually.

Something in between these lines...

Other than this, the info is very helpful, thank you.

**Comment by Nyholm** [comment]
I've updated the PR to use a better PCNTL example and changed the text to be more clear

File: components/lock.rst [hunk 1]
```diff
@@ -199,6 +199,38 @@ This component also provides two useful methods related to expiring locks:
...
+         sleep(30);
+    } else {
+         // Child process
+         echo 'The lock will be released now.'
+         exit(0);
+    }
+    // ...
+
+
+To disable this behavior, one needs to set the 3 argument to the ``LockFactory::createLock()``
```
Comment (twisted1919) [inline components/lock.rst]: Type here, i believe it should be `3rd`?

**Comment by twisted1919** [comment]
Looks okay to me, I pointed out a small typo as well, maybe you'll fix that as well.  
Thank you :)

File: components/lock.rst (hunk 1 above)
Comment (Nyholm) [reply components/lock.rst]: Thank you

File: components/lock.rst [hunk 2]
```diff
@@ -199,6 +199,38 @@ This component also provides two useful methods related to expiring locks:
...
+    } else {
+         // Child process
+         echo 'The lock will be released now.'
+         exit(0);
+    }
+    // ...
+
+
+To disable this behavior, one needs to set the 3rd argument to the ``LockFactory::createLock()``
+to false. That will make the Lock acquired for 3600 seconds or until ``Lock::released()``
```
Comment (yethee) [inline components/lock.rst]: Seems, typo in method name. Should be `Lock::release()`

Comment (Nyholm) [reply components/lock.rst]: Yes. You are correct. Thank you.

Comment (Nyholm) [reply components/lock.rst]:
```suggestion
to false. That will make the Lock acquired for 3600 seconds or until ``Lock::release()``
```

**Comment by javiereguiluz** [comment]
Tobias, thanks for this contribution and for providing a comprehensive example too.

**Comment by Nyholm** [comment]
Im happy I got all the help from @twisted1919 and the review from @yethee

**Pull request merged**